### DISCUSSION TITLE
Texlab timeout on auto format for large files

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [LazyVim/LazyVim#4226] [Q&A]
I'm using `lazyvim.plugins.extras.lang.tex`. On some longer `tex` files there is a noticeable lag on saving, followed by the warning  `notify.warn [LSP][texlab] timeout`, and no auto formatting occurs. If I toggle autoformat off, then save is immediate and there is no warning (and of course also no auto formatting).

Has anyone else experienced this? If so, what have you done? Are there good alternatives for auto formatting for `tex`? If not, what is the right way to disable autoformatting only for `tex` files in LazyVim?

**Comment by begbaj** [comment]
I have this same problem, did you find any fix for it?

**Reply by a-torgovitsky** [reply]
I didn't...seems to be a texlab issue.

**Reply by begbaj** [reply]
For others having this same issue, I noticed that sometimes just waiting a little more may fix it. It may also depend on the size of the project. Not sure why, but it works just fine most of the time, now.

Also, one thing I'm doing now is to compile it manually (actually I wrote a bash script that watches for changes in the directory and compiles every time a .tex file is updated), and this seems to have solved the issue, or at least I think so.
